GOVERNANCE.md

```diff
@@ -0,0 +1,52 @@
+# Project Governance
+
+This document provides an initial governance structure for the Perses project.
+We've deliberately kept this as simple as possible for now, but 
+we expect the governance to evolve as the project grows.
+
+## Maintainers
+
+Perses maintainers have write access to projects under the Perses GitHub organization.
+They can merge their own patches or patches from others. The current maintainers
+can be found in [MAINTAINERS.md](./MAINTAINERS.md).  Maintainers collectively
+manage the project's resources and contributors.
+
+This privilege is granted with some expectation of responsibility: maintainers
+are people who care about the Perses project and want to help it grow and
+improve. A maintainer is not just someone who can make changes, but someone who
+has demonstrated their ability to collaborate with the team, get the most
+knowledgeable people to review code and docs, contribute high-quality code, and
+follow through to fix issues (in code or tests).
+
+A maintainer is a contributor to the project's success and a citizen helping
+the project succeed.
+
+## Becoming a Maintainer
+
+To become a Maintainer you need to demonstrate the following:
+
+  * commitment to the project:
+    * participate in discussions, contributions, code and documentation reviews
+    * perform reviews for several non-trivial pull requests,
+    * contribute to several non-trivial pull requests and have them merged,
+  * ability to write quality code and/or documentation,
+  * ability to collaborate with the team,
+  * understanding of how the team works (policies, processes for testing and code review, etc),
+  * understanding of the project's code base and coding and documentation style.
+
+A new Maintainer must be proposed by an existing maintainer by creating a new issue in 
```

Comment:
Not saying it's the perfect way to do it, but as a reference, in Prometheus we require a supermajority (2/3) vote for adding new team members, and proposals are made on a private team list first in order to not embarrass anyone who doesn't receive the required number of YES votes (https://prometheus.io/governance/#team-members).

Reply:
+1 to a private maintainer-only mailing list for this. We could also use that mailing list to allow people to privately report security vulnerabilities.

I'll re-work the PR for this and some of the other feedback next week.

Reply:
I've updated this to use a private mailing list (but someone needs to create this first) :) 

I also updated it to require a majority vote for new maintainers. While I think a 2/3 vote works well for a project the size of Prometheus, with so few maintainers on a small project, we could get blocked from approving new maintainers if a couple of people go on holiday at the same time.

Reply:
> we could get blocked from approving new maintainers if a couple of people go on holiday at the same time.

I don't think it will be a problem, I don't think when a new candidate is proposed, we have to hurry to take a decision. So if people are in holidays, then the decision will take a bit longer than usual.

Reply:
I just created the google groups : perses-team@googlegroups.com :)

Reply:
@geekygirldawn So I was just discussing things with @Nexucis and we were unsure what to do about adding people like @robskillington to the team list who are not currently actively coding on Perses themselves, but are still an active part of driving the project.

My experience with this is rather limited to Prometheus and you probably have a broader view of OSS projects, so it would be interesting to get your view on this.

That said, in Prometheus we addressed that whole issue by differentiating between ["maintainers"](https://prometheus.io/governance/#team-members) (people who are actually responsible for certain pieces of code) and ["team members"](https://prometheus.io/governance/#maintainers) (which can also include people who never code, but do important community work, docs, or other valuable things that hold the project together). And the official voting body is the set of team members, with the mailing list for private votes/discussions being `prometheus-team`. For Prometheus that seems to work well at least, as it allows valuing important project contributors beyond the code maintainership level. I'm tempted to suggest that for Perses as well, but again, you've probably seen more OSS projects and how they manage that.

Reply:
We could certainly do something like this for Perses, but we would need to add details about how team members are selected and make a few other changes to what I have so far. 

I'm curious how team members are selected / approved for Prometheus?

Since a lot of people involved in Perses are also involved in Prometheus, adopting similar structures for governance and other processes makes sense to me.

Reply:
> I'm curious how team members are selected / approved for Prometheus?

So https://prometheus.io/governance/#team-members has the full details on that, but basically existing team members can nominate and vote on new ones (2/3 required). Votes are held on the private team list so as not to embarrass anyone in case it doesn't work out.

Reply:
Thanks, that's really helpful! I'll look at it in more detail make a few adjustments to this PR to adopt something similar for Perses.

Reply:
Thank you :) Yeah, for the Prometheus governance at least I can say that plenty of people have labored over every word of it when we created it to try and think of all the edge cases, so it often has some thoughtful bits. Thanos has adopted it in a slightly modified form (https://thanos.io/tip/thanos/governance.md/#governance).

Reply:
I agree that the Prometheus governance is solid and well thought out. I didn't consider using it as a model because Prometheus is so big and Perses is still tiny, but I really like the idea of incorporating what we can and doing something similar.

Fun fact, Josh Berkus recently mentioned that Prometheus provided some of the inspiration for the [CNCF Subprojects Governance Template](https://github.com/cncf/project-template/blob/main/GOVERNANCE-subprojects.md) and the Konveyor governance :)

Reply:
Actually, let me make some changes first, and then we can get reviews from a few more people.

Reply:
Alright :). Thank you @geekygirldawn !

Reply:
@geekygirldawn any news around this topic ?